src/view/BoardView.java

```diff
@@ -232,8 +249,24 @@ public boolean isCastle(ChessButton clickedButton) {
     }
 
     public void propertyChange(PropertyChangeEvent e) {
-        previousMove.clear();
-        this.previousMove = null;
+            try {
```

Comment:
rather than using a try block, I think you can use if statements and determine which propertyChange is being fired based on the name of the event that you give in the view model. The screenshot below shows an example of this from CA coding.

![Screenshot 2023-11-29 at 12 03 09 PM](https://github.com/capbloo/csc207project/assets/126191156/d72ae86c-69e7-4e01-83cc-ce5996e91933)